cli-tests/src/quarantine.rs

```diff
@@ -0,0 +1,108 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use crate::utils::{
+    generate_mock_codeowners, generate_mock_git_repo, generate_mock_valid_junit_xmls, CARGO_RUN,
+};
+use api::{
+    CreateBundleUploadRequest, CreateBundleUploadResponse, GetQuarantineBulkTestStatusRequest,
+    QuarantineConfig,
+};
+use assert_cmd::Command;
+use axum::{extract::State, Json};
+use lazy_static::lazy_static;
+use predicates::prelude::*;
+use tempfile::tempdir;
+use test_utils::mock_server::{MockServerBuilder, SharedMockServerState};
+
+// NOTE: must be multi threaded to start a mock server
+#[tokio::test(flavor = "multi_thread")]
+async fn quarantines_tests_regardless_of_upload() {
+    let temp_dir = tempdir().unwrap();
+    generate_mock_git_repo(&temp_dir);
+    generate_mock_valid_junit_xmls(&temp_dir);
+    generate_mock_codeowners(&temp_dir);
+
+    let mut mock_server_builder = MockServerBuilder::new();
+    lazy_static! {
+        static ref CALLED_COUNT: AtomicUsize = AtomicUsize::new(0);
+    }
+    mock_server_builder.set_get_quarantining_config_handler(
+        |Json(get_quarantine_bulk_test_status_request): Json<
+            GetQuarantineBulkTestStatusRequest,
+        >| {
+            let mut test_ids = get_quarantine_bulk_test_status_request
+                .test_identifiers
+                .into_iter()
+                .map(|t| t.id)
+                .collect::<Vec<_>>();
+            let called_count = CALLED_COUNT.load(Ordering::SeqCst);
+            let quarantined_tests = match called_count {
+                0 => Vec::new(),
+                1 => test_ids.split_off(1),
+                2..=3 => test_ids,
```

Comment:
ngl mixed feelings about this test structure setup. At this scale, it's very easy to see what this test is doing (since you've added the comments below), but I worry that as it scales/drifts, it will become readable. Any thoughts here or other input? I don't have a better solution, so certainly won't block on this, but I think merits mini discussion

Reply:
Nah, it's a good comment. I think a much better way would be to use an enum that's stored in this static reference. I'll update it before running each command so that it's much more declarative.